src/Core/src/Platform/Windows/ViewExtensions.cs

```diff
@@ -46,14 +46,13 @@ public static void UpdateVisibility(this FrameworkElement nativeView, IView view
 		public static void UpdateClip(this FrameworkElement nativeView, IView view)
 		{
 			var clipGeometry = view.Clip;
-
 			if (clipGeometry == null)
 				return;
 
-			if (Application.Current is not MauiWinUIApplication app)
+			if (view.Handler?.MauiContext?.Window is not Window window)
```

Comment:
I wonder if these should be called `NativeWindow` on the IMauiContext

Reply:
ah yeah... that makes more sense.

Reply:
I'll also do that in a separate PR as it involves a few changes.